api/admin/controller.py

```diff
@@ -1147,15 +1150,30 @@ def collections(self):
         if protocol and protocol not in [p.get("name") for p in protocols]:
             return UNKNOWN_PROTOCOL
 
+        name = flask.request.form.get("name")
+
         is_new = False
-        collection = get_one(self._db, Collection, name=name)
+        collection = None
+        if id:
+            collection = get_one(self._db, Collection, id=id)
+            if not collection:
+                return MISSING_COLLECTION
+
         if collection:
             if protocol != collection.protocol:
                 return CANNOT_CHANGE_PROTOCOL
-
+            if name != collection.name:
+                collection_with_name = get_one(self._db, Collection, name=name)
+                if collection_with_name:
+                    return COLLECTION_NAME_ALREADY_IN_USE
+                
         else:
+            collection_with_name = get_one(self._db, Collection, name=name)
+            if collection_with_name:
+                return COLLECTION_NAME_ALREADY_IN_USE
+
             if protocol:
-                collection, is_new = get_one_or_create(
+                collection, is_new = create(
```

Comment:
Can you call get_one_or_create and then return COLLECTION_NAME_ALREADY_IN_USE if is_new is False?